torchmultimodal/models/video_vqvae.py

```diff
@@ -4,10 +4,161 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from typing import Any, Dict, List, Tuple, Union
+
 from torch import nn, Tensor
 from torchmultimodal.modules.layers.attention import AxialAttentionBlock
 
-from torchmultimodal.modules.layers.conv import SamePadConv3d
+from torchmultimodal.modules.layers.conv import SamePadConv3d, SamePadConvTranspose3d
+
+
+class VideoEncoder(nn.Module):
+    """Encoder for Video VQVAE. Stacks specified number of ``SamePadConv3d`` layers
+    followed by a stack of ``AttentionResidualBlocks`` and a final ``SamePadConv3d``
+    layer before the codebook. The residual blocks use Axial Attention to enhance
+    representations of video data without significantly increasing computational
+    cost. Follows VideoGPT's implementation:
+    https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
+
+    Args:
+        in_channels (Tuple[int]): tuple of input channel dimension for each conv layer
```

Comment:
call it in_channels_dim? same for out

Reply:
I'm hesitant to change that, since I've kept this consistent across all components all the way down to the conv layers